src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -2193,6 +2193,21 @@ void FullCleanup()
     /// </remarks>
     void Cleanup()
     {
+        if (_stream is SslStream sslStream)
+        {
+            try
+            {
+                // Send close_notify TLS alert to correctly close connection on postgres's side
+                sslStream.ShutdownAsync().GetAwaiter().GetResult();
```

Comment:
I wonder if the comment above ("This method doesn't actually perform any meaningful I/O, and therefore is sync-only.") is still completely accurate, and whether at some point we should consider having a CleanupAsync...

Reply:
Yeah, agree it's unlikely to be an actual problem anywhere... But maybe we should do a proper async variant at some point.

Reply:
Updated the documentation for the method and opened #6012 to track changing the method to be completely async.